client/web/src/repo/settings/components/ExternalServiceEntry.tsx

```diff
@@ -0,0 +1,116 @@
+
+import { ExternalServiceCard } from '../../../components/externalServices/ExternalServiceCard'
+import { defaultExternalServices } from '../../../components/externalServices/externalServices'
+import {
+    ExcludeRepoFromExternalServiceResult,
+    ExcludeRepoFromExternalServiceVariables,
+    SettingsAreaExternalServiceFields,
+    SettingsAreaRepositoryFields,
+} from '../../../graphql-operations'
+import { EXCLUDE_REPO_FROM_EXTERNAL_SERVICE } from '../backend'
+
+import styles from './ExternalServiceEntry.module.scss'
+
+interface ExternalServiceEntryProps extends Pick<RouteComponentProps, 'history'> {
+    repo: SettingsAreaRepositoryFields
+    service: SettingsAreaExternalServiceFields
+    excludingDisabled: boolean
+    /**
+     * excludingLoading is true when there is a GraphQL mutation running.
+     * This means that all the "exclude repo" buttons (except for the button which triggered the mutation)
+     * should be disabled.
+     */
+    excludingLoading: boolean
+    /**
+     * Function to update `excludingLoading` state
+     *
+     * @param exclusionInProgress true if GraphQL mutation excluding the repo is in progress.
+     */
+    updateExclusionLoading: (exclusionInProgress: boolean) => void
+}
+
+export const ExternalServiceEntry: FC<ExternalServiceEntryProps> = ({
+    repo,
+    service,
+    excludingDisabled,
+    excludingLoading,
+    updateExclusionLoading,
+    history,
+}) => {
+    const [ttl, setTtl] = useState<number | undefined>(3)
+    const [excludeRepo, { error, data, loading: isExcluding }] = useMutation<
+        ExcludeRepoFromExternalServiceResult,
+        ExcludeRepoFromExternalServiceVariables
+    >(EXCLUDE_REPO_FROM_EXTERNAL_SERVICE, {
+        onCompleted: () => {
+            setTimeout(() => setTtl(2), 1000)
+            setTimeout(() => setTtl(1), 2000)
+            setTimeout(() => history.push(`/site-admin/external-services/${service.id}`), 3000)
+        },
+    })
+
+    return (
+        <div className={styles.grid} key={service.id}>
+            <div className={classNames(styles.card, data ? '' : 'mb-3')}>
+                <ExternalServiceCard
+                    {...defaultExternalServices[service.kind]}
+                    kind={service.kind}
+                    title={service.displayName}
+                    shortDescription="Update this code host configuration to manage repository mirroring."
+                    to={`/site-admin/external-services/${service.id}`}
+                    toIcon={null}
+                    bordered={false}
+                />
+                {error && <ErrorAlert error={`Failed to exclude repository: ${renderError(error)}`} />}
+                {data && (
+                    <Alert variant="success">
+                        {`Code host configuration updated. You will be redirected in ${ttl}...`}
+                    </Alert>
+                )}
+            </div>
+            {service.supportsRepoExclusion && !(data && !error) && (
```

Comment:
This can be simplified to the following I think:

```suggestion
            {service.supportsRepoExclusion && !data && error && (
```

Reply:
I guess it simplifies the way it looks, but ruins the meaning of this thing.

What I currently read is that "we DON'T show the button when there is a response without an error".

Disjoining data and error and putting negate operator in front of data and joining it with OR with error is complicated even for me to justify what's going on 😅